zebrad/src/components/mempool/downloads.rs

```diff
@@ -315,6 +315,27 @@ where
         Ok(())
     }
 
+    /// Cancel download/verification tasks of transactions with the
+    /// given transaction hash (see [`UnminedTxId::mined_id`]).
+    pub fn cancel(&mut self, mined_id: &transaction::Hash) {
+        // TODO: this requires going through the entire list of running tasks.
+        // If this becomes an issue, another HashMap may be needed.
+        // TODO: this can be simplified with [`HashMap::drain_filter`] which
+        // is currently nightly-only experimental API.
+        let removed_txids: Vec<UnminedTxId> = self
+            .cancel_handles
+            .keys()
+            .filter(|txid| txid.mined_id() == *mined_id)
```

Comment:
Recommended performance improvement:

#2765 makes sure that `mined_id`s are unique in the mempool, because transactions with the same `transaction::Hash` spend the same inputs. (This is true for both v4 and v5 transactions.)

So we can stop at the first match:

```suggestion
            .find(|txid| txid.mined_id() == *mined_id)
```

This reduces the inner loop iterations by about half, and also avoids allocating a vector for each outer loop iteration.

Reply:
Oh gosh you're absolutely right!

Thanks for catching this.